packages/oslo-converter-uml-ea/lib/connector-normalisation-cases/SelfAssociationWithNameConnectorCase.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  EaConnector,
+  DataRegistry,
+  NormalizedConnector,
+  EaObject,
+  EaTag,
+} from '@oslo-flanders/ea-uml-extractor';
+import { IConnectorNormalisationCase } from '../interfaces/IConnectorNormalisationCase';
+import { inject, injectable } from 'inversify';
+import { TagNames } from '../enums/TagNames';
+import {
+  getTagValue,
+  toCamelCase,
+  toPascalCase,
+  updateNameTag,
+} from '../utils/utils';
+import { Logger } from '@oslo-flanders/core';
+import { EaUmlConverterServiceIdentifier } from '../config/EaUmlConverterServiceIdentifier';
+
+@injectable()
+export class SelfAssociationWithNameConnectorCase
+  implements IConnectorNormalisationCase
+{
+  @inject(EaUmlConverterServiceIdentifier.Logger)
+  public readonly logger!: Logger;
+
+  /**
+   * Case: the connector is a self-association with a name and a source and/or destination cardinality
+   * @param connector The connector to normalise
+   * @param dataRegistry The registry to use to extract relevant information from other elements
+   * @returns An array of normalised connectors
+   */
+  public async normalise(
+    connector: EaConnector,
+    dataRegistry: DataRegistry
+  ): Promise<NormalizedConnector[]> {
+    if (
+      connector.name === null ||
+      connector.sourceObjectId !== connector.destinationObjectId
+    ) {
+      return [];
+    }
+
+    const disambiguate =
+      connector.sourceCardinality !== null &&
+      connector.destinationCardinality !== null;
+
+    const normalisedConnectors: NormalizedConnector[] = [];
+
+    const createNormalisedConnector = (
+      connector: EaConnector,
```

Comment:
Similar remark as earlier. Confusing variable name with `connector` since it's already defined on rule 34. Not an issue, since scoped